src/main/java/com/example/java/maven/ticketVendingMachine/CoinStorage.java

```diff
@@ -0,0 +1,86 @@
+package com.example.java.maven.ticketVendingMachine;
+
+import java.util.Map;
+import java.util.Optional;
+
+public class CoinStorage {
+    private final Map<Coin, Integer> coins;
+    private final MessagePrinter messagePrinter;
+
+    public CoinStorage(Map<Coin, Integer> coins, MessagePrinter messagePrinter) {
+        this.coins = coins;
+        this.messagePrinter = messagePrinter;
+    }
+
+    public Map<Coin, Integer> getCoins() {
+        return coins;
+    }
+
+    public void clear() {
+        coins.clear();
+    }
+
+    public void addCoin(Coin coin) {
+        int currentNumberOfCoins = Optional.ofNullable(coins.get(coin)).orElse(0);
+        coins.put(coin, currentNumberOfCoins + 1);
+    }
+
+    public static void moveCoinsFromTempToMainCoinStorage(CoinStorage mainCoinStorage, CoinStorage tempCoinStorage) {
```

Comment:
first of all, this method should not be static.

second, what do you think about signature: `public void moveCointsTo(CoinStorage destination)` or `public void transferCoinsTo(CoinStorage destination)` ?

third: why are you iterating through all possible coins? you don't need this `if` statement. You can iterate only through coins from storage which you move the coins - `keySet()`?

fourth: instead of using `clear()` you can get number of coins using method: `remove()` - it will remove coin from map and return value (number of this coins in map)

Reply:
Wrote new non-static method `moveCoinsTo(CoinStorage destinationStorage)`, which is much simplier, but when I tried using `remove()` instead of `clear()`, ConcurrentModificationException was thrown. Tried to use `remove()` with `Iterator`, but it was more complicated and I couldn't handle it that way and decided to stay with `clear()`.

Reply:
(Y)